app/app.go

```diff
@@ -451,28 +468,58 @@ func GetMaccPerms() map[string][]string {
 	return modAccPerms
 }
 
-func ExportAppState(name string, app *SifchainApp, ctx sdk.Context) {
-	appState, vallist, err := app.ExportAppStateAndValidators(true, []string{})
-	if err != nil {
-		ctx.Logger().Error(fmt.Sprintf("failed to export app state: %s", err))
-		return
-	}
-	appStateJSON, err := app.cdc.MarshalJSON(appState)
-	if err != nil {
-		ctx.Logger().Error(fmt.Sprintf("failed to marshal application genesis state: %s", err.Error()))
-		return
-	}
-	valList, err := json.MarshalIndent(vallist, "", " ")
-	if err != nil {
-		ctx.Logger().Error(fmt.Sprintf("failed to marshal application genesis state: %s", err.Error()))
+// RegisterAPIRoutes registers all application module routes with the provided
+// API server.
+func (app *SifchainApp) RegisterAPIRoutes(apiSvr *api.Server, apiConfig config.APIConfig) {
+	clientCtx := apiSvr.ClientCtx
+	rpc.RegisterRoutes(clientCtx, apiSvr.Router)
+	// Register legacy tx routes.
+	authrest.RegisterTxRoutes(clientCtx, apiSvr.Router)
+	// Register new tx routes from grpc-gateway.
+	authtx.RegisterGRPCGatewayRoutes(clientCtx, apiSvr.GRPCGatewayRouter)
+	// Register new tendermint queries routes from grpc-gateway.
+	tmservice.RegisterGRPCGatewayRoutes(clientCtx, apiSvr.GRPCGatewayRouter)
+
+	// Register legacy and grpc-gateway routes for all modules.
+	ModuleBasics.RegisterRESTRoutes(clientCtx, apiSvr.Router)
+	ModuleBasics.RegisterGRPCGatewayRoutes(clientCtx, apiSvr.GRPCGatewayRouter)
+
+	// register swagger API from root so that other applications can override easily
+	if apiConfig.Swagger {
+		RegisterSwaggerAPI(clientCtx, apiSvr.Router)
 	}
+}
 
-	err = ioutil.WriteFile(fmt.Sprintf("%v/%v-state.json", DefaultNodeHome, name), appStateJSON, 0600)
-	if err != nil {
-		ctx.Logger().Error(fmt.Sprintf("failed to write state to file: %s", err.Error()))
-	}
-	err = ioutil.WriteFile(fmt.Sprintf("%v/%v-validator.json", DefaultNodeHome, name), valList, 0600)
+// RegisterTxService implements the Application.RegisterTxService method.
+func (app *SifchainApp) RegisterTxService(clientCtx client.Context) {
+	authtx.RegisterTxService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.BaseApp.Simulate, app.interfaceRegistry)
+}
+
+// RegisterTendermintService implements the Application.RegisterTendermintService method.
+func (app *SifchainApp) RegisterTendermintService(clientCtx client.Context) {
+	tmservice.RegisterTendermintService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.interfaceRegistry)
+}
+
+// RegisterSwaggerAPI registers swagger route with API Server
+func RegisterSwaggerAPI(ctx client.Context, rtr *mux.Router) {
+	statikFS, err := fs.New()
 	if err != nil {
-		ctx.Logger().Error(fmt.Sprintf("failed to write Validator List to file: %s", err.Error()))
+		panic(err)
```

Comment:
Do we really want to panic here instead of just logging an error?

Reply:
I would say panics are fine here because this code gets called at the start of a node/chain. It would not panic in the middle of running the process.